File: frontend/src/services/MenuService.js
import axios from '../utils/axiosConfig';

class MenuService {
  /**
   * 获取所有菜单
   */
  getAllMenus(languageCode = null) {
    let url = '/api/menus';
    if (languageCode) {
      url += `?languageCode=${languageCode}`;
    }
    return axios.get(url);
  }

  /**
   * 根据ID获取菜单
   */
  getMenuById(id) {
    return axios.get(`/api/menus/${id}`);
  }

  /**
   * 创建菜单
   */
  createMenu(menu) {
    return axios.post('/api/menus', menu);
  }

  /**
   * 更新菜单
   */
  updateMenu(id, menu) {
    return axios.put(`/api/menus/${id}`, menu);
  }

  /**
   * 删除菜单
   */
  deleteMenu(id) {
    return axios.delete(`/api/menus/${id}`);
  }

  /**
   * 批量删除菜单
   */
  batchDeleteMenus(ids) {
    return axios.delete('/api/menus/batch', { data: ids });
  }

  /**
   * 导出菜单
   */
  exportMenus() {
    return axios.get('/api/menus/export');
  }

  /**
   * 导入菜单
   */
  importMenus(menus, override = false) {
    return axios.post(`/api/menus/import?override=${override}`, menus);
  }

  /**
   * 刷新菜单缓存
   */
  refreshMenuCache() {
    return axios.post('/api/menus/refresh-cache');
  }

  /**
   * 获取菜单操作日志
   */
  getMenuOperationLogs() {
    return axios.get('/api/menus/operation-logs');
  }

  /**
   * 根据菜单ID获取操作日志
   */
  getMenuOperationLogsByMenuId(menuId) {
    return axios.get(`/api/menus/${menuId}/operation-logs`);
  }
  
  /**
   * 获取角色菜单权限
   */
  getRoleMenuPermissions(roleId) {
    return axios.get(`/api/roles/${roleId}/menus`);
  }
  
  /**
   * 保存角色菜单权限
   */
  saveRoleMenuPermissions(params) {
    return axios.post(`/api/roles/${params.roleId}/menus`, params.menuIds);
  }
}

export default new MenuService();

File: src/main/java/com/itheima/activiti/service/MenuService.java
package com.itheima.activiti.service;

import com.itheima.activiti.entity.Permission;
import org.springframework.cache.annotation.CacheEvict;
import org.springframework.cache.annotation.CachePut;
import org.springframework.cache.annotation.Cacheable;

import java.util.List;

/**
 * 菜单服务接口
 */
public interface MenuService {
    
    /**
     * 查询所有菜单，支持按语言代码查询
     */
    List<Permission> getAllMenus(String languageCode);
    
    /**
     * 根据ID查询菜单
     */
    Permission getMenuById(Long id);
    
    /**
     * 创建菜单
     */
    Permission createMenu(Permission menu);
    
    /**
     * 更新菜单
     */
    Permission updateMenu(Permission menu);
    
    /**
     * 删除菜单
     */
    boolean deleteMenu(Long id);
    
    /**
     * 批量删除菜单
     */
    boolean batchDeleteMenus(List<Long> ids);
    
    /**
     * 导出菜单数据
     */
    List<Permission> exportMenus();
    
    /**
     * 导入菜单数据
     */
    boolean importMenus(List<Permission> menus, boolean override);
    
    /**
     * 刷新菜单缓存
     */
    void refreshMenuCache();
}
